pkg/proxy/filter_endpoints_test.go

```diff
@@ -17,16 +17,34 @@ limitations under the License.
 package proxy
 
 import (
-	"reflect"
+	"fmt"
 	"testing"
 
 	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	featuregatetesting "k8s.io/component-base/featuregate/testing"
 	"k8s.io/kubernetes/pkg/features"
 )
 
+func checkExpectedEndpoints(expected sets.String, actual []Endpoint) error {
```

Comment:
I'm on the fence about whether this function is simpler or easier to understand than the reflect.DeepEqual check that exists today

Reply:
sure, _this function_ is more confusing than using reflect.DeepEqual, but I think it makes the actual test cases simpler, because it makes it so that `expectedEndpoints` only contains the information we are actually checking. Without this change, the "merge tests into a single test" commit would have resulted in changing the `expectedEndpoints` values for some of the tests even though we were still selecting the same set of endpoints. And likewise, future changes that made us look at additional input data when filtering the endpoints would result in having to change the "expected" results to match, even in the cases where the added input data didn't change the set of endpoints that was selected.